AdyenCashAppPay/CashAppPayComponent.swift

```diff
@@ -0,0 +1,260 @@
+
+    internal lazy var storeDetailsItem: FormToggleItem = {
+        let storeDetailsItem = FormToggleItem(style: configuration.style.toggle)
+        storeDetailsItem.title = localizedString(.cardStoreDetailsButton, configuration.localizationParameters)
+        storeDetailsItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: ViewIdentifier.storeDetailsItem)
+    
+        return storeDetailsItem
+    }()
+
+    internal lazy var cashAppPayButton: FormButtonItem = {
+        let item = FormButtonItem(style: configuration.style.mainButtonItem)
+        item.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: ViewIdentifier.cashAppButtonItem)
+        item.title = localizedString(.cashAppPayTitle, configuration.localizationParameters)
+        item.buttonSelectionHandler = { [weak self] in
+            self?.didSelectSubmitButton()
+        }
+        return item
+    }()
+
+    // MARK: - Private
+
+    private lazy var formViewController: FormViewController = {
+        let formViewController = FormViewController(style: configuration.style)
+        formViewController.delegate = self
+        formViewController.localizationParameters = configuration.localizationParameters
+        formViewController.title = paymentMethod.displayInformation(using: configuration.localizationParameters).title
+    
+        if configuration.showsStorePaymentMethodField {
+            formViewController.append(storeDetailsItem)
+        }
+    
+        formViewController.append(FormSpacerItem(numberOfSpaces: 2))
+        formViewController.append(cashAppPayButton)
+
+        return formViewController
+    }()
+
+    /// Initializes the Cash App Pay component.
+    ///
+    /// - Parameter paymentMethod: The Cash App Pay  payment method.
+    /// - Parameter context: The context object for this component.
+    /// - Parameter configuration: The configuration for the component.
+    public init(paymentMethod: CashAppPayPaymentMethod,
+                context: AdyenContext,
+                configuration: CashAppPayConfiguration) {
+        self.cashAppPayPaymentMethod = paymentMethod
+        self.context = context
+        self.configuration = configuration
+    }
+
+    private func didSelectSubmitButton() {
+        guard formViewController.validate() else { return }
+    
+        startLoading()
+        startCashAppPayFlow()
+    }
+
+    private func startLoading() {
+        cashAppPayButton.showsActivityIndicator = true
+        formViewController.view.isUserInteractionEnabled = false
+    }
+
+    public func stopLoading() {
+        cashAppPayButton.showsActivityIndicator = false
+        formViewController.view.isUserInteractionEnabled = true
+    }
+
+    private func startCashAppPayFlow() {
+        let actions = createPaymentActions()
+        guard actions.isEmpty == false else {
+            AdyenAssertion.assertionFailure(message: "At least one paymentAction is required to create a CashAppPay rquest")
```

Comment:
This error is not very descriptive:
Is it really clear to merchant what is "paymentAction" ?
How can one know that "oneTimePayment" was not created, because `payment` was not set to AdyenContext or that amount is zero?

Reply:
Because they may only be going for pre auth with no amount and only set up recurring. In that case only onFile action would be needed and configuration needs to be set.
But checking those error cases in more details seemed redundant. It's only 2 element array and not very hard to see which one can be the cause. 

Open to discuss this one as well :)

Reply:
How about making this checks in init like we do for ApplePay?
This will prevent showing it on DropIn as well

Reply:
We can't move it to init since actions are created after the button tap.